### DISCUSSION TITLE
Netlify Functions and support for $app/dynamic/*

### DISCUSSION THREAD

**Discussion by kyllerss** [sveltejs/kit#8128] [Q&A]
Netlify background functions seem to run on a more traditional Node deployment that doesn't have access to some SvelteKit features. I have a single Netlify background function that needs to do database activity, etc.., and would like to use SvelteKit's  `$app/dynamic/*` to simplify my usage of env variables. 

However, when I invoke the Netlify background function, I get the following error:

```
docker-netlify-functions-1  | ✘ [ERROR] Could not resolve "$env/dynamic/private"
docker-netlify-functions-1  | 
docker-netlify-functions-1  |     src/lib/server/env/env_serverside.ts:5:20:
docker-netlify-functions-1  |       5 │ import { env } from "$env/dynamic/private";
docker-netlify-functions-1  |         ╵                     ~~~~~~~~~~~~~~~~~~~~~~
```

The error message recommends that I mark it as an external module in the `netlify.toml` file, but that just results in an error stating that the module could not be found during runtime. 

Any suggestions as to what I should look into?

**Comment by MentalGear** [comment]
Could you explain how you got background fncs working with netlify (there's no word on it in the guides) ?

**Reply by kyllerss** [reply]
It's not pretty:

1. Any code that is directly or indirectly imported by the background function has to revert to this pattern: 
```
return process.env.MY_SECRET ?? import.meta.env.MY_SECRET;
```
2. At one point early on I ran into issues where some classes that I was importing didn't have the `process.env` values initialized and would blow up when imported (I haven't looked at it in some time - I could have been wrong in what was causing the issue). The pattern I adopted was to initialize these classes in the handler code. For example in my case, I had to initialize database connections explicitly as the first step within the handler code.

Much of these issues I ran into as I was learning SvelteKit + Netlify - so my solutions are likely not the best.

Hope this helps!
